### DISCUSSION TITLE
SP montar o nfeProc, juntando o retorno com a NFe

### DISCUSSION THREAD

**Discussion by valdecirsanches** [TadaSoftware/PyNFe#265] [General]
Alguem está autorizando nfe em SP estou com dificuldade de pegar o xml após autorizado... não entendo o processo... se puderem me explicar ou me enviar parte do codigo que autoriza e constroe o nfe.xml agradeceria muito.

**Comment by valdecirsanches** [comment]
Não é mais necessário ja fiz a montagem

def montaxml(nf): # nf é o registro da tabela que tem guardado o path do XML "NFE" e o protocolo de autorização junta-se os dois

    xml_file = nf.xml_file
    f = open(xml_file, encoding='utf8')
    xmlnfe = f.read()
    f.close()

    x = nf.xml_autorizacao
    f = open(x, encoding='utf8')
    xmlaut = f.read()
    f.close()

    obj = xmltodict.parse(xmlaut)
    aut = xmlaut[xmlaut.find('<protNFe'):xmlaut.find('</protNFe>') + 10]

    raiz = etree.Element('nfeProc', xmlns=NAMESPACE_NFE, versao=VERSAO_PADRAO)
    raiz.append(etree.fromstring(xmlnfe))
    raiz.append(etree.fromstring(aut))

    xxx = etree.tostring(raiz)
    xxx = xxx.decode("utf-8")

    arquivo = open(xml_file, 'wb')
    arquivo.write(str(xxx).encode())
    arquivo.close()

    nf.xml_file = xml_file
    nf.save()

    return
    # Mais ou menos issu ai!!!!

**Reply by leogregianin** [reply]
@valdecirsanches, após agregar a NFe com o protocolo você conferiu o XML se a assinatura digital está válida pelo site https://www.sefaz.rs.gov.br/nfe/nfe-val.aspx ?
